motionplan/motionPlanner.go

```diff
@@ -107,6 +109,21 @@ func PlanWaypoints(ctx context.Context,
 	fs frame.FrameSystem,
 	worldState *frame.WorldState,
 	motionConfigs []map[string]interface{},
+) ([]map[string][]frame.Input, error) {
+	return motionPlanInternal(ctx, logger, goals, f, seedMap, fs, worldState, motionConfigs)
+}
+
+// motionPlanInternal is the internal private function that all motion planning access calls. This will construct the plan manager for each
+// waypoint, and return at the end.
+// This has the same function signature as `PlanWaypoints` but is a private function so as to not have public functions call other.
```

Comment:
I'm not sure why this function is being made, looks like the comment isn't completed? If we can avoid adding it I would appreciate that, as it is we have a ton of endpoints for calling motion related things and this will add another one with its own nuanced reasons for calling it over others

Reply:
Comment is complete but is missing a word, should be "each other" at the end.

> it is we have a ton of endpoints for calling motion related things and this will add another one with its own nuanced reasons for calling it over others

This does the opposite of that. This is a private function so a user can't call it. The public functions are public, and should not be called by anything internal. It's not an external endpoint, and it reduces internal endpoint possibilities from four to one.

Having an internal shared function will be more maintainable, and prevents having cascading issues if we alter one public function.

Reply:
See here for another place where we use this paradigm, where a more complex piece of functionality is reused:

https://github.com/viamrobotics/goutils/blob/main/rpc/dial.go

Reply:
Yeah, I think this is a good call. On the other hand, does this mean we're supporting too many public facing functions here? Not something we should consider too much in the context of this PR, but `PlanFrameMotion` existing to support motion planning without a frame system, when we've really thrown our design behind leveraging the Frame System MORE, does seem like it will eventually be deprecated.

Regardless, I think this is better from an internal support standpoint. This should have zero impact on the user-facing experience (if anything, could be argued to be a net positive).

Reply:
The use case of `PlanFrameMotion` is the entrypoint for arm motion planning in the case where the arm is not a part of a framesystem. This may indeed get deprecated (or we could move it to `arm` if we're worried about entrypoint count on motion).

Reply:
Acknowledged. Let's circle back on this when we get some more user feedback on the experience here.

Reply:
Ok I see I can get behind it.  Is it more or less idiomatic/clear to name it `planMotion` instead?